### DISCUSSION TITLE
Remove useless named arguments

### DISCUSSION THREAD

**Discussion by lowwa132** [PHP-CS-Fixer/PHP-CS-Fixer#7373] [Feature ideas]
As stated in this comment https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/issues/6627#issuecomment-1712548444 named parameters used in a bad way impact readability, for example:

```php
array_map(callback: static fn ($id) => (int) $id, array: array_column(array: $query->getArrayResult(), column_key: 'id'));
```

Is it possible to consider having a detection of such cases, and remove useless named arguments when it corresponds to function's signature ?

**Comment by Wirone** [comment]
Hi @lowwa132 🙂. I believe Fixer is not a good place for that because we run analysis without full SCA context. We just tokenise the code and run rules against tokens. When analysing function call, we don't have any information about its signature (unless it's defined in the same file), so we can't determine if named arguments are superfluous or not.

I believe tools like PHPStan or Psalm fit this case better 🙂.

**Reply by lowwa132** [reply]
Hi @Wirone , thank you for your input, I opened an issue https://github.com/phpstan/phpstan/issues/10074
